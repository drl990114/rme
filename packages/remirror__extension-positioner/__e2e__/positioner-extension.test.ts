import { getDocument, queries } from 'playwright-testing-library';
import { ElementHandle } from 'playwright-testing-library/dist/typedefs';
import { goto, typist } from 'testing/playwright';

describe('Positioner Extension', () => {
  const { getByRole, getByTestId, getByText } = queries;
  let $document: ElementHandle;
  let $editor: ElementHandle;

  describe('cursorPositioner', () => {
    const path = 'positioner-extension--cursor-positioner';

    beforeEach(async () => {
      await goto(path);
      $document = await getDocument(page);
      $editor = await getByRole($document, 'textbox');
    });

    it('should show the bubble menu', async () => {
      await $editor.focus();
      await typist('This is text');
      await expect($editor.innerHTML()).resolves.toMatchSnapshot();
      const $bubbleMenu = await getByTestId($document, 'bubble-menu');
      await expect($bubbleMenu.getAttribute('style')).resolves.toBe(
        'bottom:999999px;left:-999999px;position:absolute',
      );

      await typist('{selectall}');
      const $visibleBubbleMenu = await getByTestId($document, 'bubble-menu');
      const newStyles = await $visibleBubbleMenu.getAttribute('style');
      expect(newStyles).toInclude('bottom');
      expect(newStyles).not.toInclude('999999px');
      expect(newStyles).toInclude('left');

      const $boldButton = await getByText($document, 'Bold');
      await $boldButton.click();

      await expect($editor.innerHTML()).resolves.toMatchSnapshot();
    });
  });
});
