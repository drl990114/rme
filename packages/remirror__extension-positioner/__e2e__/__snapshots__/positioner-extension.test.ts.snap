// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Positioner Extension cursorPositioner should show the bubble menu 1`] = `
<span data-id="remirror-positioner-widget"
      role="presentation"
      contenteditable="false"
      class="ProseMirror-widget"
>
  <span class="css-f0r864"
        style="top: 19px; left: 94.1875px; width: 1px; height: 18px; background: blue;"
  >
  </span>
  <div id="id-1"
       role="dialog"
       tabindex="-1"
       data-dialog="true"
       class="css-1k97hgp"
       style
  >
    <button data-testid="bubble-menu-bold">
      Bold
    </button>
    <button>
      Italic
    </button>
    <button>
      Underline
    </button>
  </div>
</span>
<p class>
  This is text
</p>
`;
